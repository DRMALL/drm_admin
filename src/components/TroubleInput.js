import React from 'react'
import Input from 'antd/lib/input'
import Select from 'antd/lib/select'
import getInputValue from '../actions/getInputValue'
// import { troubleType } from '../commons/troubleCommon'
import dispatch from '../actions/dispatch'
import changeTroubleType from '../utils/changeTroubleType'
import LzEditor from 'react-lz-editor'
import beforeUpload from '../utils/beforeUpload'
import  { troubleImg } from '../commons/apis'
import Editor from '../containers/Editor'


export default props => {
  const { title, categoryxx, content, troubleKinds, isEdit, troubleImgList } = props.state.trouble
  function getValue(e){
    console.log(e)
    dispatch('TROUBLE_GET_CONTENT_VALUE', { content: e })
  }
  const token = localStorage.getItem('token')
  // const uploadProps = {
  //     action: `${troubleImg}?token=${token}`,
  //     name: 'bugs',
  //     onChange: onChange,
  //     listType: 'picture',
  //     fileList: troubleImgList,
  //     data: (file) => {
  //     },
  //     multiple: true,
  //     beforeUpload: beforeUpload,
  //     showUploadList: true
  // }
  // function onChange(e){
  //   let result = e.fileList
  //   result.map((item, index )=> {
  //     item.url = item.response ? item.response.data.url  : ''
  //     item.key = '-' + index
  //   })
  //   dispatch('TROUBLE_GET_RICH_TEXT_IMGAGES', result)
  // }

  return (
    <div>
      <div className='trouble-input-flex' >
        <div className='trouble-input-left' >标题</div>
        <Input style={{ width:300 }} onChange={ getInputValue }
               data-path='TROUBLE' data-id='title'
               placeholder='输入标题' value={ title } />
      </div>
      <div className='trouble-input-flex' >
        <div className='trouble-input-left' >分类</div>
        <Select style={{ width:300 }} onChange={ troubleSelect }
               placeholder='请输入分类' value={ changeTroubleType(categoryxx, troubleKinds) }  >
          {
            troubleKinds.map( (item, index) =>
            <Select.Option key={ index } value={ item._id }>{item.text}</Select.Option> )
          }
        </Select>
      </div>
      <div className='trouble-input-flexs' >
        <div  className='trouble-input-lefts' >问题描述</div>
        <div className='trouble-input-editor'>
{/*        <LzEditor cbReceiver={ getValue }  video={ false } uploadProps={uploadProps}
                audio={ false } urls={ false } importContent={ content }  active={isEdit}
                autoSave={ false }/>*/}

          <Editor bounds='.trouble-input-editor' 
                  onChange={getValue}
                  value={content}/>
        </div>
      </div>
    </div>
    )
}

function troubleSelect(value){
  // console.log(value)
  dispatch('TROUBLE_GET_SELECT_VALUE', value)
}
