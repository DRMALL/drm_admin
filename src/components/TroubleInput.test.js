

import React from 'react'
import TroubleInput from './TroubleInput'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('TroubleInput test', ()=> {
  const state = { trouble: { title:'标题', troubleKinds: [] } }
  const wrapper = render(<TroubleInput state={ state } /> )
  expect(toJson(wrapper)).toMatchSnapshot()
})
