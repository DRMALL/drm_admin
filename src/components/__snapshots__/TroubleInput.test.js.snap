// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TroubleInput test 1`] = `
<div>
  <div
    class="trouble-input-flex"
  >
    <div
      class="trouble-input-left"
    >
      标题
    </div>
    <input
      class="ant-input"
      data-id="title"
      data-path="TROUBLE"
      placeholder="输入设备编号"
      style="width:300px;"
      type="text"
      value="标题"
    />
  </div>
  <div
    class="trouble-input-flex"
  >
    <div
      class="trouble-input-left"
    >
      分类
    </div>
    <div
      class="ant-select ant-select-enabled"
      style="width:300px;"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
                  ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection__placeholder"
            style="display:none;user-select:none;-webkit-user-select:none;"
            unselectable="unselectable"
          >
            压力分类字典
          </div>
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1;"
            title=""
          />
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none;"
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    class="trouble-input-flexs"
  >
    <div
      class="trouble-input-lefts"
    >
      设备描述
    </div>
    <textarea
      class="ant-input"
      data-id="content"
      data-path="TROUBLE"
      placeholder="输入对设备的描述"
      style="width:300px;"
    />
  </div>
</div>
`;
